File: AdSurgeDemo/AdSurgeDemo/AdSurgeDemoUI/RewardedViewController.m
//
//  RewardedViewController.m
//  AdSurgeDevDemo
//
//  Created by katie on 2025/4/17.
//

#import <Foundation/Foundation.h>

#import <AdSurgeSDK/AdSurgeRewardedAd.h>

#import "AdLogTable.h"
#import "RewardedViewController.h"

@interface RewardedViewController ()<AdSurgeRewardedAdDelegate>

@property (nonatomic) AdSurgeRewardedAd *rewardedAd;
@property (nonatomic, strong) AdLogTable *logTable;

@end

@implementation RewardedViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.unitIdTitles = @[
        @"General 10856",
    ];
    self.unitIds = @[
        @"10856",
    ];
    self.unitId = self.unitIds.firstObject;
    self.logTable = [[AdLogTable alloc] initWithParentView:self.view withTopMargin:20];
}

- (void)loadAd {
    self.rewardedAd = [[AdSurgeRewardedAd alloc] initWithAdUnitIdentifier:self.unitId];
    self.rewardedAd.delegate = self;
    
    AdSurgeAdConfig *adConfig = [AdSurgeAdConfig configWithAdFormat:AdSurgeAdFormat.rewarded];
    adConfig.mute = [SettingManager sharedManager].isMuted;
    [self.rewardedAd loadAd:adConfig];
}

- (void)setNavigationBar {
    [super setNavigationBar];
    UILabel *label = [[UILabel alloc] init];
    label.text = @"RewardAD";
    label.textColor = [UIColor whiteColor];
    label.textAlignment = NSTextAlignmentCenter;
    label.font = [UIFont boldSystemFontOfSize:18];
    
    [label sizeToFit];
    self.navigationItem.titleView = label;
}

- (void)showBtnTapped:(UIButton *)sender {
    [super showBtnTapped:sender];
    if (![self.rewardedAd isValid]) {
        [self.logTable addLogWithInfo:@"Ad not valid" ad:nil error:nil reward:nil];
        return;
    }
    [self.rewardedAd showAdFromRootViewController:self];
}

#pragma mark - AdSurgeRewardedAdDelegate

- (void)didLoadAd:(AdSurgeAd *)ad {
    NSLog(@"%s",__FUNCTION__);
    NSLog(@"ad's format: %@, adUnitIdentifier: %@, revenue: %f",ad.format.label, ad.adUnitIdentifier, ad.revenue);
    [self.logTable addLogWithInfo:NSStringFromSelector(_cmd) ad:ad error:nil reward:nil];
}

- (void)didFailToLoadAdForAdUnitIdentifier:(NSString *)adUnitIdentifier withError:(AdSurgeError *)error {
    NSLog(@"%s",__FUNCTION__);
    NSLog(@"ad's adUnitIdentifier: %@, error: %@",adUnitIdentifier, error);
    [self.logTable addLogWithInfo:NSStringFromSelector(_cmd) ad:nil error:error reward:nil];
}

- (void)didDisplayAd:(AdSurgeAd *)ad withError:(AdSurgeError *)error {
    NSLog(@"%s \n",__FUNCTION__);
    NSLog(@"%@ \n",error);
    [self.logTable addLogWithInfo:NSStringFromSelector(_cmd) ad:ad error:error reward:nil];
}

- (void)didHideAd:(AdSurgeAd *)ad {
    NSLog(@"%s",__FUNCTION__);
    [self.logTable addLogWithInfo:NSStringFromSelector(_cmd) ad:ad error:nil reward:nil];
}

- (void)didClickAd:(AdSurgeAd *)ad {
    NSLog(@"%s",__FUNCTION__);
    [self.logTable addLogWithInfo:NSStringFromSelector(_cmd) ad:ad error:nil reward:nil];
}

- (void)didRewardUserForAd:(AdSurgeAd *)ad withReward:(AdSurgeReward *)reward {
    NSLog(@"%s",__FUNCTION__);
    [self.logTable addLogWithInfo:NSStringFromSelector(_cmd) ad:ad error:nil reward:reward];
}

- (void)didPayRevenueForAd:(AdSurgeAd *)ad {
    NSLog(@"%s",__FUNCTION__);
    [self.logTable addLogWithInfo:NSStringFromSelector(_cmd) ad:ad error:nil reward:nil];
}

@end

File: AdSurgeDemo-Swift/AdSurgeDemo-Swift/AdSurgeDemoUI/RewardedViewController.swift
//
//  RewardedViewController.swift
//  AdSurgeDemo-Swift
//
//  Created by katie on 2025/6/5.
//

import Foundation
import UIKit
import AdSurgeSDK

class RewardedViewController: BaseAdViewController, AdSurgeRewardedAdDelegate {

    var rewardedAd: AdSurgeRewardedAd!

    override func viewDidLoad() {
        super.viewDidLoad()
        self.unitIdTitles = [
            "General 10856",
        ];
        self.unitIds = [
            "10856",
        ]
        self.unitId = unitIds.first
    }

    override func setNavigationBar() {
        let label = UILabel()
        label.text = "RewardAD"
        label.textColor = .white
        label.textAlignment = .center
        label.font = UIFont.boldSystemFont(ofSize: 18)
        label.sizeToFit()
        self.navigationItem.titleView = label
    }

    override func loadAd() {
        rewardedAd = AdSurgeRewardedAd(adUnitIdentifier: self.unitId ?? "10856")
        rewardedAd?.delegate = self

        let adConfig = AdSurgeAdConfig(adFormat: .rewarded)
        adConfig.mute = SettingManager.shared.isMuted
        rewardedAd.load(adConfig)
    }
    
    override func showBtnTapped(_ sender: UIButton) {
        super.showBtnTapped(sender)
        if (!rewardedAd.isValid) {
            addCell("Ad not valid")
            return
        }
        rewardedAd.show(fromRootViewController: self)
    }

    // MARK: - AdSurgeRewardedAdDelegate

    func didLoad(_ ad: AdSurgeAd) {
        print(#function)
        print("ad's format: \(ad.format.label), adUnitIdentifier: \(ad.adUnitIdentifier), revenue: \(ad.revenue)")
        addCell(#function)
    }

    func didFailToLoadAd(forAdUnitIdentifier adUnitIdentifier: String, withError error: AdSurgeError) {
        print(#function)
        print("ad's adUnitIdentifier: \(adUnitIdentifier), error: \(error)")
        addCell(#function)
    }

    func didDisplay(_ ad: AdSurgeAd, withError error: AdSurgeError?) {
        print(#function)
        print("\(String(describing: error))")
        addCell(#function)
    }

    func didHide(_ ad: AdSurgeAd) {
        print(#function)
        addCell(#function)
    }

    func didClick(_ ad: AdSurgeAd) {
        print(#function)
        addCell(#function)
    }

    func didRewardUser(for ad: AdSurgeAd, with reward: AdSurgeReward) {
        print(#function)
        addCell(#function)
    }

    func didPayRevenue(for ad: AdSurgeAd) {
        print(#function)
        addCell(#function)
    }
}
